### DISCUSSION TITLE
Integrating MontePy with VS Code Extension

### DISCUSSION THREAD

**Discussion by SterlingButters** [idaholab/MontePy#343] [Ideas]
Again, sorry I'm not following the feature request template since this may be out of scope/impossible. I think this project's usefulness would be compounded by integration into a VS Code Extension e.g. https://github.com/repositony/vscode_mcnp. Since VS Code Extensions are developed in typescript, it might be a challenge to bridge the gap but thought I would throw the idea out there.

**Comment by MicahGale** [comment]
So what features are you looking for in general beyond what both of these libraries provide independently?

More practically, I think this is out of the scope of this repo specifically, and should rather be use MontePy as a dependency for such a tool. Adding support for n+1 text editors in MontePy directly would be a bad idea. (If we support VScode, I would add vim first personally.)

If someone wants to add this tool fantastic, but I personally won't do it, because JavaScript is one of my least favorite language. As for complexity. Python to type-script libraries must exist right?

**Reply by SterlingButters** [reply]
The advantage of a VSCode Extension is the provision of a GUI - so to answer your question explicitly, a GUI feature is what I would be looking for. This would permit MCNP users without python proficiency to take advantage of the package tools. Even those that _are_ proficient in python might benefit from interaction in a GUI which is possibly faster than writing code (depending on the complexity of the operation(s)). While I'm proficient enough in python myself, I'm speaking from the standpoint of extending your user base beyond the programmatically inclined. 

I could get by in TypeScript but I by no means enjoy it since my proficiency in Python is far superior (so I won't be contributing in this capacity either). 

> Python to type-script libraries must exist right?

It appears they do: https://github.com/marcj/pybridge  
For anyone of adequate skillset, I wouldn't see it as particularly difficult to bind MontePy functions/methods to graphic elements of a VSCode extension. Maybe the resources exist at INL for this?

**Reply by MicahGale** [reply]
Ok I can see this use case. I still maintain that MontePy is not the place for this directly. It's meant to be a lightweight Python library for the input files that is easy to install. That's why it only has two dependencies, none of which requires the end user to compile anything. Adding a GUI integration would add a lot of dependencies that not all users would use. So if someone wants to create a wrapper that imports MontePy I am all for it.

**Reply by SterlingButters** [reply]
I think you had it right the first time: that MontePy would be a dependency for the VS Code Extension - not necessarily creating a GUI around MontePy (which would create a number of dependencies). Maybe if I ever got the time one day I might look more into it
